packages/edit-post/src/editor.js

```diff
@@ -136,7 +131,7 @@ function Editor( {
 			// all block types).
 			const defaultAllowedBlockTypes =
 				true === settings.allowedBlockTypes
-					? map( blockTypes, 'name' )
+					? blockTypes.map( ( { name } ) => name )
```

Comment:
Will always be an array:

https://github.com/WordPress/gutenberg/blob/ec2bdc59c4c4cdf8d86cdd432ee30eaca6f14eed/packages/blocks/src/store/selectors.js#L71